### PR TITLE
Add option for using webc v3

### PR DESCRIPTION
**Pull request by maminrayej**
This PR adds support for the `USE_V3` env var. If set, the cli will download `v3` version of a package.

### CODE REVIEW

**Comment by ayys** [comment]
btw I'd call the env var `WASMER_USE_WEBCV3`

**Comment by maminrayej** [comment]
> btw I'd call the env var `WASMER_USE_WEBCV3`

You're right. That's better.

File: lib/wasix/src/runtime/resolver/wapm_source.rs [hunk 1]
```diff
@@ -343,15 +343,36 @@ fn decode_summary(
...
+            WapmWebQueryGetPackageVersionDistribution {
+                webc_version: v3_webc_version,
+                pirita_sha256_hash: v3_pirita_sha256_hash,
+                pirita_download_url: v3_pirita_download_url,
             },
         ..
     } = pkg_version;
 
+    let (webc_version, pirita_sha256_hash, pirita_download_url) =
+        if std::env::var("WASMER_USE_WEBCV3").is_ok() {
```
Comment (syrusakbary) [inline lib/wasix/src/runtime/resolver/wapm_source.rs:362]: This needs to be a configuration variable that is passes to WASIX. WASIX should NOT access env vars, only the CLI should

**Review by syrusakbary — DISMISSED** [review]
Address env var comment

**Review by theduke — CHANGES_REQUESTED** [review]

File: lib/cli/src/commands/run/mod.rs [hunk 2]
```diff
@@ -107,11 +107,20 @@ impl Run {
             wasmer_vm::set_stack_size(self.stack_size.unwrap());
         }
 
+        // check for the preffered webc version
+        let preffered_webc = match std::env::var("WASMER_USE_WEBCV3") {
```
Comment (syrusakbary) [inline lib/cli/src/commands/run/mod.rs:111]: This naming has a typo (two f's)... also I think we can use `preferred_webc_version` which is a bit more descriptive

File: lib/cli/src/commands/run/wasi.rs [hunk 3]
```diff
@@ -533,6 +533,7 @@ impl Wasi {
         engine: Engine,
         env: &WasmerEnv,
         rt_or_handle: I,
+        preffered_webc: webc::Version,
```
Comment (syrusakbary) [inline lib/cli/src/commands/run/wasi.rs:536]:
```suggestion
        preferred_webc_version: webc::Version,
```

File: lib/wasix/src/runtime/resolver/wapm_source.rs [hunk 4]
```diff
@@ -275,13 +286,14 @@ impl Source for WapmSource {
...
 
-        matching_package_summaries(response, &version_constraint)
+        matching_package_summaries(response, &version_constraint, self.preffered_webc)
     }
 }
 
 fn matching_package_summaries(
     response: WapmWebQuery,
     version_constraint: &VersionReq,
+    preffered_webc: webc::Version,
```
Comment (syrusakbary) [inline lib/wasix/src/runtime/resolver/wapm_source.rs:296]:
```suggestion
        preferred_webc_version: webc::Version,
```

File: lib/cli/src/commands/run/wasi.rs [hunk 5]
```diff
@@ -643,7 +645,8 @@ impl Wasi {
         let graphql_endpoint = self.graphql_endpoint(env)?;
         let cache_dir = env.cache_dir().join("queries");
         let mut wapm_source = WapmSource::new(graphql_endpoint, Arc::clone(&client))
-            .with_local_cache(cache_dir, WAPM_SOURCE_CACHE_TIMEOUT);
+            .with_local_cache(cache_dir, WAPM_SOURCE_CACHE_TIMEOUT)
+            .with_prefferred_webc(preffered_webc);
```
Comment (syrusakbary) [inline lib/cli/src/commands/run/wasi.rs:649]:
```suggestion
            .with_preferred_webc_version(preffered_webc);
```

File: lib/wasix/src/runtime/resolver/wapm_source.rs [hunk 6]
```diff
@@ -26,6 +26,7 @@ pub struct WapmSource {
     client: Arc<dyn HttpClient + Send + Sync>,
     cache: Option<FileSystemCache>,
     token: Option<String>,
+    preffered_webc: webc::Version,
```
Comment (syrusakbary) [inline lib/wasix/src/runtime/resolver/wapm_source.rs:29]:
```suggestion
    preferred_webc_version: webc::Version,
```

File: lib/wasix/src/runtime/resolver/wapm_source.rs [hunk 7]
```diff
@@ -340,18 +352,40 @@ fn decode_summary(
...
-                pirita_download_url,
+                webc_version: v3_webc_version,
+                pirita_sha256_hash: v3_pirita_sha256_hash,
+                pirita_download_url: v3_pirita_download_url,
             },
         ..
     } = pkg_version;
 
+    let (webc_version, pirita_sha256_hash, pirita_download_url) =
+        if preffered_webc == webc::Version::V3 {
```
Comment (syrusakbary) [inline lib/wasix/src/runtime/resolver/wapm_source.rs:375]: Improve naming

File: lib/wasix/src/runtime/resolver/wapm_source.rs [hunk 8]
```diff
@@ -340,18 +352,40 @@ fn decode_summary(
     namespace: &str,
     package_name: &str,
     pkg_version: WapmWebQueryGetPackageVersion,
+    preffered_webc: webc::Version,
```
Comment (syrusakbary) [inline lib/wasix/src/runtime/resolver/wapm_source.rs:355]: Improve naming

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**